### PR TITLE
#165305763 fix following/unfollow/get followers/get following status  messages

### PR DESCRIPTION
**Pull request by Amoswachira**
#### What does this PR do?
Add descriptive error messages to the follow, unfollow, get followers and get following endpoints  then return the  following field.
#### Description of Task to be completed?
make sure the following,followers unfollow response messages return proper error messages
### Local Development Setup
 - First Create python virtual env
 ```
  $ virtualenv -p python .venv
 ```
 - Install Requirements
 ```
  $ pip install -r requiremts.txt
 ```
 - Copy .env-example to .env and set config
 ```
  $ cp .env-example .env
 ```
 - Create postgres database
 ```
  $ sudo su postgres
  $ psql
  postgres=# CREATE USER django WITH PASSWORD 'password';
  postgres=# ALTER ROLE django SET client_encoding TO 'utf8';
  postgres=# ALTER ROLE django SET default_transaction_isolation TO 'read committed';
  postgres=# ALTER ROLE django SET timezone TO 'UTC';
  postgres=# CREATE DATABASE drf;
  postgres=# GRANT ALL PRIVILEGES ON DATABASE drf TO django;
  postgres=# \q
  $ exit
````
clone the project
```
git clone https://github.com/andela/ah-the-immortals-backend.git
```
**After cloning the repo:-**
```
$ git checkout  ft-fix-follow-messages-165305763

 ```
- Run Server

  $ python manage.py runserver

-Then test the endpoints using swagger of postman

 

#### What are the relevant pivotal tracker stories?

[#165305763](https://www.pivotaltracker.com/n/projects/2326460/stories/165305763)

#### Screenshots (if appropriate)
screencapture-127-0-0-1-8000-api-swagger-2019-05-05-22_21_52

### CODE REVIEW

**Review by matthenge — CHANGES_REQUESTED** [review]
Fix merge conflicts

**Comment by Amoswachira** [comment]
@matthenge thanks let me fix them

**Review by Ogutu-Brian — APPROVED** [review]
LGTM

**Review by matthenge — APPROVED** [review]

**Review by matthenge — APPROVED** [review]
LGTM

**Review by Ogutu-Brian — APPROVED** [review]

File: authors/apps/profiles/views.py [hunk 1]
```diff
@@ -188,6 +215,8 @@ def get(self, request, username):
                 "first_name": profile.first_name,
                 "last_name": profile.last_name,
                 "bio": profile.bio,
+                "following": True,
```
Comment (oluwagbenga-joloko) [inline authors/apps/profiles/views.py]: The following field should not be hardcoded. Does this mean I follow all users that follow me?  can you implement a custom function to check the following status here?

Comment (Amoswachira) [reply authors/apps/profiles/views.py]: @oluwagbenga-joloko okay let me work on that

Comment (Amoswachira) [reply authors/apps/profiles/views.py]: @oluwagbenga-joloko I  have added  a function to check for the following status

File: authors/apps/profiles/tests/test_follow.py [hunk 2]
```diff
@@ -39,7 +44,7 @@ def test_followed_Successfully(self):
             '/api/profiles/Thanos/follow/')
         self.assertEqual(res.status_code, status.HTTP_202_ACCEPTED)
         self.assertEqual(json.loads(res.content), {'profile': {
-            'username': 'Thanos', 'first_name': '', 'last_name': '', 'bio': ''}})
+            'username': 'Thanos', 'first_name': '', 'last_name': '', 'bio': '', 'following': True}})
```
Comment (3N61N33R) [inline authors/apps/profiles/tests/test_follow.py]: Why are you hard coding this?

File: authors/apps/profiles/views.py [hunk 3]
```diff
@@ -83,13 +85,20 @@ def patch(self, request, username):
         )
 
 
-class FollowAPI(GenericAPIView):
+class UserNotFound(APIException):
+    """exception message"""
+    status_code = 404
+    default_detail = 'User with that username Not found'
+
```
Comment (3N61N33R) [inline authors/apps/profiles/views.py]: save exceptions in an exceptions file to make it reusable

**Review by 3N61N33R — COMMENTED** [review]
Hey @Amoswachira 
Add the command for cloning the repo as well on your PR

File: authors/apps/profiles/views.py (hunk 3 above)
Comment (Amoswachira) [reply authors/apps/profiles/views.py]: @graycadeau thank you for the feedback let me work on that

File: authors/apps/profiles/tests/test_follow.py (hunk 2 above)
Comment (Amoswachira) [reply authors/apps/profiles/tests/test_follow.py]: @graycadeau let me fix that

**Comment by oluwagbenga-joloko** [comment]
@Amoswachira please ensure the `following` field is added when getting a list of profiles and when getting a single profile. It should not be added when a user tries to get his own profile though.

File: authors/apps/profiles/views.py (hunk 3 above)
Comment (Amoswachira) [reply authors/apps/profiles/views.py]: @graycadeau I have made the exception file reusable

File: authors/apps/profiles/tests/test_follow.py (hunk 2 above)
Comment (Amoswachira) [reply authors/apps/profiles/tests/test_follow.py]: @graycadeau I have removed the hard coding

**Comment by Amoswachira** [comment]
@graycadeau i have added the clone command

**Comment by Amoswachira** [comment]
@oluwagbenga-joloko let me work on that

**Comment by Amoswachira** [comment]
@oluwagbenga-joloko  I completed the changes you asked for

**Pull request merged**